projects/igniteui-angular/src/lib/combo/combo.common.ts

```diff
@@ -27,6 +53,1244 @@ export interface IgxComboBase {
+            this.valid = IgxComboState.INITIAL;
+        }
+        this.manageRequiredAsterisk();
+    };
+
+    /**
+     * Constructs the combo display value
+     * If remote, caches the key displayText
+     * If not, just combine the object.displayKeys
+     */
+    protected createDisplayText(newSelection: any[], oldSelection: any[]) {
+        let value = '';
+        if (this.isRemote) {
+            if (newSelection.length) {
+                const removedItems = oldSelection.filter(e => newSelection.indexOf(e) < 0);
+                const addedItems = newSelection.filter(e => oldSelection.indexOf(e) < 0);
+                this.registerRemoteEntries(addedItems);
+                this.registerRemoteEntries(removedItems, false);
+                value = Object.keys(this._remoteSelection).map(e => this._remoteSelection[e]).join(', ');
+            } else {
+                // If new selection is empty, clear all items
+                this.registerRemoteEntries(oldSelection, false);
+            }
+        } else {
+            value = this.concatDisplayText(newSelection);
+        }
+        return value;
+    }
+
+    protected checkMatch(): void {
+        const itemMatch = this.filteredData.some(this.findMatch);
+        this.customValueFlag = this.allowCustomValues && !itemMatch;
+    }
+
+    protected findMatch = (element: any): boolean => {
+        const value = this.displayKey ? element[this.displayKey] : element;
+        return value.toString().toLowerCase() === this.searchValue.trim().toLowerCase();
+    };
+
+    protected manageRequiredAsterisk(): void {
+        if (this.ngControl && this.ngControl.control.validator) {
+            // Run the validation with empty object to check if required is enabled.
+            const error = this.ngControl.control.validator({} as AbstractControl);
+            this.inputGroup.isRequired = error && error.required;
+        }
+    }
+
+    /** Contains key-value pairs of the selected valueKeys and their resp. displayKeys */
+    private registerRemoteEntries(ids: any[], add = true) {
+        if (add) {
+            const selection = this.getValueDisplayPairs(ids);
+            for (const entry of selection) {
+                this._remoteSelection[entry[this.valueKey]] = entry[this.displayKey];
+            }
+        } else {
+            for (const entry of ids) {
+                delete this._remoteSelection[entry];
+            }
+        }
+    }
+
+    /**
+     * For `id: any[]` returns a mapped `{ [combo.valueKey]: any, [combo.displayKey]: any }[]`
+     */
+    private getValueDisplayPairs(ids: any[]) {
+        return this.data.filter(entry => ids.indexOf(entry[this.valueKey]) > -1).map(e => ({
+            [this.valueKey]: e[this.valueKey],
+            [this.displayKey]: e[this.displayKey]
+        }));
+    }
+
+    /** Returns a string that should be populated in the combo's text box */
+    private concatDisplayText(selection: any[]): string {
```

Comment:
do we need this in common? IMO simple combo does not need such method.

Reply:
Actually it's used in `createDisplayText` which is protected and is being used in the simple combo.